### PR TITLE
Simplify `localStorage` usage

### LINKED ISSUE #1512: Use the `useLocalStorage` composable from VueUse for reactive localStorage binding
https://vueuse.org/core/useLocalStorage/

We use [localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) for saving user preferences. However it is not reactive so we have to jump through some hoops such as duplicating values in the Vuex store which is reactive.

Using this composable looks like it will simplify our usage of localStorage. However it also adds a new dependency, though this might not be much of a problem due to tree-shaking.

### PR DESCRIPTION
**Pull request by MetRonnie**
Closes #1512 

Use VueUse's [`useLocalStorage()`](https://vueuse.org/core/useLocalStorage/) composable to simplify how we use `localStorage`.

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Tests are included
- [x] No changelog entry as not user-facing
- [x] No docs

### CODE REVIEW

**Review by markgrahamdawson — APPROVED** [review]
I've done some manual testing and looks like it works as expected

File: src/mixins/appSettings.js [hunk 1]
```diff
@@ -23,9 +25,10 @@ export default {
      * @param {boolean} value
      */
     setReducedAnimation (value) {
-      localStorage.reducedAnimation = value
-      this.$store.commit('app/setReducedAnimation', value)
+      useReducedAnimation().value = value
       for (const property of ['transition', 'ripple']) {
+        // TODO: can use useDefaults() composable when we upgrade to Vuetify 3.2+
+        // and convert this mixin into a composable
```
Comment (oliver-sanders) [inline src/mixins/appSettings.js:31]:
```suggestion
        // TODO: can use useDefaults() composable when we upgrade to Vuetify 3.2+
        // and convert this mixin into a composable
        // https://github.com/cylc/cylc-ui/issues/1551
```

File: src/views/Workspace.vue [hunk 2]
```diff
@@ -78,7 +79,7 @@ if (import.meta.env.MODE !== 'production') {
   )
 }
 
-export const defaultView = () => localStorage.defaultView || 'Tree'
+export const defaultView = () => useLocalStorage('defaultView', 'Tree')
```
Comment (oliver-sanders) [inline src/views/Workspace.vue:82]: Should this get moved into `localStorage.js` with the others?

**Review by oliver-sanders — APPROVED** [review]

File: renovate.json [hunk 3]
```diff
@@ -47,6 +47,11 @@
       "matchPackageNames": ["@unhead/vue"],
       "schedule": ["on the 3rd day of the month"]
     },
+    {
+      "matchPackageNames": ["@vueuse/core"],
+      "automerge": true,
```
Comment (MetRonnie) [inline renovate.json:52]: I am happy with automerging as the localStorage tests should be comprehensive enough to catch any problems

**Pull request merged**